File: sdk/src/types/usage.ts
/** A small preview of the Agents referencing a resource — `agentCount` is the real total. */
export interface ResourceUsage {
  /** The real total count of referencing Agents — use this, not `agents.length`. */
  agentCount: number;
  /** A preview list, capped at 20 — not necessarily every matching Agent when `agentCount > 20`. */
  agents: Array<{ _id: string; name: string }>;
}

File: sdk/python/src/personaai/types/usage.py
"""Mirrors ``sdk/src/types/usage.ts``."""

from __future__ import annotations

from typing import TypedDict


class ResourceUsageAgent(TypedDict):
    _id: str
    name: str


class ResourceUsage(TypedDict):
    """A small preview of the Agents referencing a resource — ``agentCount`` is the real total."""

    agentCount: int  # the real total count of referencing Agents — use this, not len(agents)
    agents: list[ResourceUsageAgent]  # a preview list, capped at 20 — not necessarily every match
